clientconn.go

```diff
@@ -268,7 +257,7 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 		cc.authority = creds.Info().ServerName
 	} else if cc.dopts.insecure && cc.dopts.authority != "" {
 		cc.authority = cc.dopts.authority
-	} else if unixScheme {
+	} else if strings.HasPrefix(cc.target, "unix:") {
```

Comment:
This needs to come from the resolver (with `resolver.Address` - is this the same as `ServerName`? @menghanl?).  It can be done in a follow-up change but this is not what we want to be doing, ideally.

Reply:
So this is what I thought and what I was worried about.  The resolver needs a way to set the authority for addresses.  I think `ServerName` is the right way to do that -- WDYT?  Do we want a different handshaker name from authority header value?

Reply:
I think they should be the same. For security reason?

Reply:
Agreed; I think we can consider it a bug that we are not using the address's ServerName for the authority header.  But this can be done in a follow-up PR.